src/frontend/src/common/queries/auth.ts

```diff
@@ -55,15 +55,16 @@ export function useUserInfo(): UseQueryResult<UserResponse, unknown> {
   });
 }
 
-export function useProtectedRoute(): UseQueryResult<UserResponse, unknown> {
-  const router = useRouter();
-  const result = useUserInfo();
-
+export function protectedRoute(
+  result: UseQueryResult<UserResponse, unknown>,
+  router: NextRouter
+): void {
```

Comment:
have to pass result and router in like so because we're wrapping this function in a useEffect hook (only way we can use next JS router to push to new route if not wrapped in button link (client side redirecting)). Or at least that was my understanding of this issue. Would be nice to run by @mmmmmmaya and @tihuan to verify.

Reply:
I'm not an expert here, so I would love to hear @tihuan's opinion